bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -542,7 +542,8 @@ protected ChannelFuture connect() {
         try {
             addr = bookieAddressResolver.resolve(bookieId);
         } catch (BookieAddressResolver.BookieIdNotResolvedException err) {
-            LOG.error("Cannot connect to {} as endpopint resolution failed", bookieId, err);
+            LOG.error("Cannot connect to {} as endpoint resolution failed (probably bookie is down)",
```

Comment:
I'm a bit confused about the `probably bookie is down` part. Why resolving the address requires the bookie to be up? I'm probably missing something basic.

Reply:
this "resolution" is about looking up the "current" Bookie network address on ZooKeeper, it is not DNS resolution.

When the bookie is up, it advertises all of its open endpoints (bookie-rpc,http....) on ZK, in the same znode that we are using to detect if the bookie is writable/readonly, it is an ephemeral znode.

When you want to connect to a bookie, you start from a BookieID (an opaque string), you lookup the current network address on ZK (we have a cache of course) and then you are good to go.

but if the Bookie is down the ephemeral znode does not exist and the client does not know the network address to connect to.

Usually when you see this log line the problem is that you cannot connect to the bookie because you do not know the network address, because it is not advertising it, and this usually happens when the bookie is down. 

for reference:
http://bookkeeper.apache.org/bps/BP-41-bookieid/
http://bookkeeper.apache.org/bps/BP-38-bookie-endpoint-discovery/

Reply:
I can remove it, but I find it quite useful in the day to day support to people looking at this kind of logs

Reply:
Could this be a ZK error too, like ZK is down and it can't talk to ZK? If so, then perhaps add it to the message. Otherwise, it is fine to leave it there.

Reply:
That exception means that we currently cannot resolve  the is.
So I am keeping the message, according to your answer.
Basically in this API we are throwing that Exception instead of returning null.

Beside note: we are not performing blocking accesses to ZK for this operation, we are always reading from an in-memory cache